misk-redis/src/main/kotlin/misk/redis/Redis.kt

```diff
@@ -130,4 +131,64 @@ interface Redis {
    * See [incr] for extra information.
    */
   fun incrBy(key: String, increment: Long): Long
+
+  /**
+   * Set a timeout on key. After the timeout has expired, the key will automatically be deleted. A
+   * key with an associated timeout is often said to be volatile in Redis terminology.
+   *
+   * The timeout will only be cleared by commands that delete or overwrite the contents of the key,
+   * including [del], [set], GETSET and all the *STORE commands. This means that all the operations
+   * that conceptually alter the value stored at the key without replacing it with a new one will
+   * leave the timeout untouched. For instance, incrementing the value of a key with [incr], pushing
+   * a new value into a list with LPUSH, or altering the field value of a hash with [hset] are all
+   * operations that will leave the timeout untouched.
+   *
+   * The timeout can also be cleared, turning the key back into a persistent key, using the PERSIST
+   * command.
+   *
+   * If a key is renamed with RENAME, the associated time to live is transferred to the new key
+   * name.
+   *
+   * If a key is overwritten by RENAME, like in the case of an existing key Key_A that is
+   * overwritten by a call like RENAME Key_B Key_A, it does not matter if the original Key_A had a
+   * timeout associated or not, the new key Key_A will inherit all the characteristics of Key_B.
+   *
+   * Note that calling [expire]/[pExpire] with a non-positive timeout or [expireAt]/[pExpireAt] with
+   * a time in the past will result in the key being deleted rather than expired (accordingly, the
+   * emitted key event will be del, not expired).
+   *
+   * @return 1 if the timeout was set. 0 if the timeout was not set. e.g. key doesn't exist, or
+   * operation skipped due to the provided arguments.
```

Comment:
I missed this in the incr/incrBy PR, but I think a boolean return type makes more sense here based on the rest of the interface.

Ideally there's more of an indication to the caller of why the operation failed, but my guess is that the underlying library doesn't provide that?

Reply:
It doesn't seem like the underlying library provides it.

Definitely agree that a boolean return type makes more sense here, didn't see that on `del`. It doesn't make sense for the `incr`/`incrBy` code, since that returns the new value rather than a success/failure state